livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py

```diff
@@ -0,0 +1,471 @@
+    sanitized_arguments: dict[str, Any] = {}
+    for arg_info in fnc_info.arguments.values():
+        if arg_info.name not in parsed_arguments:
+            if arg_info.default is inspect.Parameter.empty:
+                raise ValueError(
+                    f"AI function {fnc_name} missing required argument {arg_info.name}"
+                )
+            continue
+
+        arg_value = parsed_arguments[arg_info.name]
+        if get_origin(arg_info.type) is not None:
+            if not isinstance(arg_value, list):
+                raise ValueError(
+                    f"AI function {fnc_name} argument {arg_info.name} should be a list"
+                )
+
+            inner_type = get_args(arg_info.type)[0]
+            sanitized_value = [
+                _sanitize_primitive(
+                    value=v, expected_type=inner_type, choices=arg_info.choices
+                )
+                for v in arg_value
+            ]
+        else:
+            sanitized_value = _sanitize_primitive(
+                value=arg_value, expected_type=arg_info.type, choices=arg_info.choices
+            )
+
+        sanitized_arguments[arg_info.name] = sanitized_value
+
+    return llm.function_context.FunctionCallInfo(
+        tool_call_id=tool_call_id,
+        raw_arguments=raw_arguments,
+        function_info=fnc_info,
+        arguments=sanitized_arguments,
+    )
+
+
+def _build_function_description(
+    fnc_info: llm.function_context.FunctionInfo,
+) -> anthropic.types.ToolParam:
+    def build_schema_field(arg_info: llm.function_context.FunctionArgInfo):
+        def type2str(t: type) -> str:
+            if t is str:
+                return "string"
+            elif t in (int, float):
+                return "number"
+            elif t is bool:
+                return "boolean"
+
+            raise ValueError(f"unsupported type {t} for ai_property")
+
+        p: dict[str, Any] = {}
+        if arg_info.default is inspect.Parameter.empty:
+            p["required"] = True
+        else:
+            p["required"] = False
+
+        if arg_info.description:
+            p["description"] = arg_info.description
+
+        if get_origin(arg_info.type) is list:
+            inner_type = get_args(arg_info.type)[0]
+            p["type"] = "array"
+            p["items"] = {}
+            p["items"]["type"] = type2str(inner_type)
+
+            if arg_info.choices:
+                p["items"]["enum"] = arg_info.choices
+        else:
+            p["type"] = type2str(arg_info.type)
+            if arg_info.choices:
+                p["enum"] = arg_info.choices
+
+        return p
+
+    input_schema: dict[str, object] = {"type": "object"}
+
+    for arg_info in fnc_info.arguments.values():
+        input_schema[arg_info.name] = build_schema_field(arg_info)
+
+    return {
+        "name": fnc_info.name,
+        "description": fnc_info.description,
+        "input_schema": input_schema,
+    }
+
+
+def _sanitize_primitive(
```

Comment:
Seems like there is no change compared to https://github.com/livekit/agents/blob/80dac5600bcc1c0863b98982a626e201f1be97a4/livekit-agents/livekit/agents/llm/_oai_api.py#L143

Reply:
yeah I can put this in a common utils in another PR. I'll hold off for now because tests are flaky (due to 429s)